### PR TITLE
Provide System.Threading.Tasks.Dataflow package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `System.Threading.Tasks.Dataflow` NuGet package readme.

See https://github.com/dotnet/runtime/pull/99358

### CODE REVIEW

**Comment by ViktorHofer** [comment]
ping @dotnet/area-system-threading-tasks

File: src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,143 @@
...
+    Console.WriteLine("Creating word list...");
+
+    // Remove common punctuation by replacing all non-letter characters with a space character.
+    text = nonLetterRegex.Replace(text, " ");
+
+    // Separate the text into an array of words.
+    return text.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+});
+
+// Removes short words and duplicates.
```
Comment (stephentoub) [inline src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md:52]: It's not removing duplicates
```suggestion
// Removes short words
```

File: src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md [hunk 2]
```diff
@@ -2,43 +2,143 @@
...
+
+// Finds all words in the specified collection whose reverse also exists in the collection.
+TransformManyBlock<string[], string> findReversedWords = new TransformManyBlock<string[], string>(words =>
+{
+    Console.WriteLine("Finding reversed words...");
+
+    var wordsSet = new HashSet<string>(words);
+
+    return from word in wordsSet
+           let reverse = new string(word.Reverse().ToArray())
```
Comment (stephentoub) [inline src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md:70]:
```suggestion
           let reverse = string.Concat(word.Reverse())
```

File: src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md [hunk 3]
```diff
@@ -2,43 +2,143 @@
...
+           where word != reverse && wordsSet.Contains(reverse)
+           select word;
+});
+
+// Prints the provided reversed words to the console.
+ActionBlock<string> printReversedWords = new ActionBlock<string>(reversedWord =>
+{
+    Console.WriteLine("Found reversed words {0}/{1}", reversedWord, new string(reversedWord.Reverse().ToArray()));
```
Comment (stephentoub) [inline src/libraries/System.Threading.Tasks.Dataflow/src/PACKAGE.md:78]:
```suggestion
    Console.WriteLine("Found reversed words {0}/{1}", reversedWord, string.Concat(reversedWord.Reverse()));
```

**Review by stephentoub — APPROVED** [review]
Thanks!

**Pull request merged**